addons/dango-rain/userscript.js

```diff
@@ -0,0 +1,100 @@
+let dangorain = false,
+  dangoContainerLeft,
+  dangoContainerRight,
+  vissiableDangos,
+  noticeTimeout;
+const div = () => document.createElement("div");
+const noticeText = div();
+noticeText.className = "sa-dango-notice";
+
+const checkForDango = (text) => {
+  text = text.value || text.textContent;
+  if (!text.includes("dango")) {
+    if (dangorain && dangoContainerLeft) {
+      dangoContainerLeft?.remove();
+      dangoContainerRight?.remove();
+      noticeText?.remove();
+    }
+    dangorain = false;
+    return;
+  }
+  if (dangorain) return;
+  dangorain = true;
+  dangoContainerLeft = div();
+  dangoContainerLeft.className = "sa-dangos-left";
+  for (let i = 0; i < 20; i++) {
+    const dango = div();
+    dango.className = "sa-dango";
+    dango.style.left = (i % 10) * 10 + "%";
+    dango.style.animationDelay = `${Math.random() * 8}s, ${Math.random() * 8}s`;
+    dangoContainerLeft.append(dango);
+  }
+  setEltWidth(dangoContainerLeft);
+
+  dangoContainerRight = dangoContainerLeft.cloneNode(true);
+  dangoContainerRight.className = "sa-dangos-right";
+
+  document.querySelector("#content").append(dangoContainerLeft, dangoContainerRight);
+  if (!localStorage.getItem("scratchAddonsAprilFoolsModal2021")) document.querySelector("#content").append(noticeText);
+};
+const setEltWidth = (elt) => {
+  if (!elt) return;
+  let sideWidth = (document.body.clientWidth - document.querySelector("#profile-data").clientWidth) / 2;
+  sideWidth = sideWidth - 25;
+  elt.style.width = sideWidth + "px";
+  let old = vissiableDangos;
+  vissiableDangos = sideWidth > 0;
+  if (vissiableDangos) {
+    clearTimeout(noticeTimeout);
+    if (!old && !localStorage.getItem("scratchAddonsAprilFoolsModal2021")) {
+      document.querySelector("#content").append(noticeText);
+    }
+  } else {
+    // Window resizing is very weird on some devices...
+    // Some devices might make the notice text "flicker".
+    clearTimeout(noticeTimeout);
+    noticeTimeout = setTimeout(() => {
+      noticeText?.remove();
+    }, 1000);
+  }
+};
+addEventListener("resize", () => {
+  setEltWidth(dangoContainerLeft);
+  setEltWidth(dangoContainerRight);
+});
+
+export default async function ({ addon, global, console, msg }) {
+  const notifClose = Object.assign(document.createElement("span"), {
+    style: `
```

Comment:
Move to css file?

Reply:
this is taken from cs.js, and also code by W_L... up to him